### PR TITLE
Feature/python configuration

### PR DESCRIPTION
**Pull request by raquel-ucl**
Coherent Python, Numpy and Behave are now installed using GreatCMakeCookOff.

### CODE REVIEW

File: DCProgsConfig.h.in [hunk 1]
```diff
@@ -42,7 +42,7 @@
 # define DCPROGS_DECL_CONSTEXPR(TYPEANDNAME, VALUE) constexpr TYPEANDNAME
 #else 
 # define DCPROGS_INIT_CONSTEXPR(TYPEANDNAME, VALUE) const static TYPEANDNAME
-# define DCPROGS_DECL_CONSTEXPR(TYPEANDNAME, VALUE) const TYPEANDNAME = VALUE
+# define DCPROGS_DECL_CONqSTEXPR(TYPEANDNAME, VALUE) const TYPEANDNAME = VALUE
```
Comment (jenshnielsen) [inline DCProgsConfig.h.in]: Looks like this was unintentional?

Comment (raquel-ucl) [inline DCProgsConfig.h.in]: Fixed!

File: cmake_modules/AllPythonBindings.cmake [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
 
-find_package(PythonLibs REQUIRED)
-find_package(PythonInterp REQUIRED)
+# Install Python with GreatCMakeCookOff, as well as numpy and behave
+include(PythonPackage)
+include(PythonPackageLookup)
+find_package(CoherentPython)
+find_python_package(numpy)
+find_package(Numpy REQUIRED)
+find_python_package(behave)
```
Comment (jenshnielsen) [inline cmake_modules/AllPythonBindings.cmake]: Does this actually do anything. We are finding behave again below. Again like sphinx this is really just being used as an executable

**Comment by raquel-ucl** [comment]
The PR test seems to be successful, but in the Legion nodes it complains about not being able to find Doxygen. Is it because it needs installing or because CMake is not working properly?

**Comment by jenshnielsen** [comment]
afaik doxygen is not on legion so that is fine

File: cmake_modules/AllPythonBindings.cmake (hunk 2 above)
Comment (raquel-ucl) [inline cmake_modules/AllPythonBindings.cmake]: You're right. Removed.

**Comment by jenshnielsen** [comment]
retest this please

**Comment by jenshnielsen** [comment]
great can you try removing the work arounds from https://github.com/UCL-RITS/jenkins-job-builder-files/blob/master/jobs/dcprogs.yml that sets `PYTHON_LIBRARY` and `PYTHON_INCLUDE_DIR` that should no longer be needed.

**Comment by raquel-ucl** [comment]
That workedin Jenkins changing directly in the job configuration in Jenkins, but can't push those changes yet because that'll make other stuff fail if I remember correctly, right?

**Comment by jenshnielsen** [comment]
yes

**Pull request merged**